var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -106,55 +106,47 @@ def generate_gpu_config(self):
 
     def cmake_args(self):
         spec = self.spec
-        options = []
-
-        options.extend(
-            [
-                "-DCMAKE_INSTALL_PREFIX=%s" % self.prefix,
-                "-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib" % self.prefix,
-                "-DBLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
-                # As of MAGMA v2.3.0, CMakeLists.txt does not use the variable
-                # BLAS_LIBRARIES, but only LAPACK_LIBRARIES, so we need to
-                # explicitly add blas to LAPACK_LIBRARIES.
-                "-DLAPACK_LIBRARIES=%s" % (spec["lapack"].libs + spec["blas"].libs).joined(";"),
-            ]
-        )
-
-        options += ["-DBUILD_SHARED_LIBS=%s" % ("ON" if ("+shared" in spec) else "OFF")]
+        define = self.define
+        options = [
+            define("CMAKE_INSTALL_PREFIX", self.prefix),
+            define("CMAKE_INSTALL_NAME_DIR:PATH", self.prefix.lib),
+            define("BLAS_LIBRARIES", [spec["blas"].libs]),
+            # As of MAGMA v2.3.0, CMakeLists.txt does not use the variable
+            # BLAS_LIBRARIES, but only LAPACK_LIBRARIES, so we need to
+            # explicitly add blas to LAPACK_LIBRARIES.
+            define("LAPACK_LIBRARIES", [spec["lapack"].libs, spec["blas"].libs]),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+        ]
 
         if spec.satisfies("%cce"):
-            options += ["-DCUDA_NVCC_FLAGS=-allow-unsupported-compiler"]
+            options.append(define("CUDA_NVCC_FLAGS", "-allow-unsupported-compiler"))
 
         if "+fortran" in spec:
-            options.extend(["-DUSE_FORTRAN=yes"])
+            options.append(define("USE_FORTRAN", True))
             if spec.satisfies("%xl") or spec.satisfies("%xl_r"):
-                options.extend(["-DCMAKE_Fortran_COMPILER=%s" % self.compiler.f77])
-
+                options.append(define("CMAKE_Fortran_COMPILER", self.compiler.f77))
             if spec.satisfies("%cce"):
-                options.append("-DCMAKE_Fortran_FLAGS=-ef")
+                options.append(define("CMAKE_Fortran_FLAGS", "-ef"))
 
         if spec.satisfies("^cuda"):
-            cuda_arch = self.spec.variants["cuda_arch"].value
-            if "@:2.2.0" in spec:
-                capabilities = " ".join("sm{0}".format(i) for i in cuda_arch)
-                options.extend(["-DGPU_TARGET=" + capabilities])
-            else:
-                capabilities = " ".join("sm_{0}".format(i) for i in cuda_arch)
-                options.extend(["-DGPU_TARGET=" + capabilities])
+            cuda_arch = spec.variants["cuda_arch"].value
+            sep = "" if "@:2.2.0" in spec else "_"
+            capabilities = " ".join("sm{0}{1}".format(sep, i) for i in cuda_arch)
```

Comment:
should this be a list? i.e. semi-colon separated string?

Reply:
The MAGMA CMakeLists.txt consistently uses spaces: https://bitbucket.org/icl/magma/src/master/CMakeLists.txt